From 2325575763b282aa2245715286e237f1b73ef549 Mon Sep 17 00:00:00 2001
Message-Id: <2325575763b282aa2245715286e237f1b73ef549.1421897934.git.chang-joon.lee@intel.com>
In-Reply-To: <149a6863e085e113eb3c4f75ec994895411447d6.1421897934.git.chang-joon.lee@intel.com>
References: <149a6863e085e113eb3c4f75ec994895411447d6.1421897934.git.chang-joon.lee@intel.com>
From: Namrta <namrta.salonie@intel.com>
Date: Tue, 18 Nov 2014 22:30:48 +0530
Subject: [PATCH 4/7] FOR_UPSTREAM [VPG]: drm/i915: Potential NULL pointer
 de-reference in ggtt unbind.

Found by static analysis tool.

v2: Fix commit msg (Rafael)

Issue: GMINL-4354
Change-Id: Ic6bce234ff6d501e77a5d6f9311616dee45705df
Signed-off-by: Namrta <namrta.salonie@intel.com>
---
 drivers/gpu/drm/i915/i915_drv.h |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_drv.h b/drivers/gpu/drm/i915/i915_drv.h
index ffad05c..8d82b12 100644
--- a/drivers/gpu/drm/i915/i915_drv.h
+++ b/drivers/gpu/drm/i915/i915_drv.h
@@ -2841,7 +2841,10 @@ i915_gem_obj_ggtt_pin(struct drm_i915_gem_object *obj,
 static inline int
 i915_gem_object_ggtt_unbind(struct drm_i915_gem_object *obj)
 {
-	return i915_vma_unbind(i915_gem_obj_to_ggtt(obj));
+	struct i915_vma *vma = i915_gem_obj_to_ggtt(obj);
+	if (!vma)
+		return -EINVAL;
+	return i915_vma_unbind(vma);
 }
 
 void i915_gem_object_ggtt_unpin(struct drm_i915_gem_object *obj);
-- 
1.7.9.5

